### PR TITLE
Feat(Authentication): Implement Base Infrastructure Required for Authentication 

### PR DESCRIPTION
**Pull request by Godhanded**
# Description
Set up Infrastructure required for Authentication and Authorization for The Boiler Plate. It Should support JWT Authentication, Password Hashing and Password Comparing, these would be used for the Signup and Register Endpoints
 
 Partially Implements issue #8 and issue #67
 
 # Changes proposed
 Generate JWT Contatining User Details, support Password Hashing And Comparison, Register Authentication Middlewares 
 
 ## What were you told to do?
Generate JWT Contatining User Details, support Password Hashing And Comparison, Register Authentication Middlewares 
 
 ## What did you do?
 * Created Password Service and Interface
 * Implemented Token Services and Interface
 * Registered Authentication Services in ApplicationConfigurations 
 * Ensured Swagger Docs support
 * Added Authentication Middleware
 
 # Check List (Check all the applicable boxes)
 * [x]  Create Password Service and Interface
 * [x]  Register Authentication Services in ApplicationConfigurations
 * [x]  Ensure Swagger Docs support
 * [x]  Implement Token Services and Interface
 * [x]   Add Authentication Middleware
 
 🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.
 
 * [x]  My code follows the code style of this project.
 * [x]  This PR does not contain plagiarized content.
 * [x]  The title and description of the PR is clear and explains the approach.
 * [x]  I am making a pull request against the **dev branch** (left side).
 * [x]  My commit messages styles matches our requested structure.
 * [x]  My code additions will fail neither code linting checks nor unit test.
 * [x]  I am only making changes to files I was requested to.

### CODE REVIEW

File: src/Hng.Infrastructure/Services/PasswordService.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        private const int SALTSIZE = 10;
+
+        public (string passwordSalt, string hashedPassword) GeneratePasswordSaltAndHash(string plainPassword)
+        {
+            var buffer = RandomNumberGenerator.GetBytes(SALTSIZE);
+            var salt = Convert.ToBase64String(buffer);
+            string hashedPassword = GeneratePasswordHash(plainPassword, salt);
+            return (salt, hashedPassword);
+        }
+
```
Comment (NdubuisiJr) [inline src/Hng.Infrastructure/Services/PasswordService.cs:18]: Remove extran new line

File: src/Hng.Web/Program.cs [hunk 2]
```diff
@@ -13,7 +14,25 @@
 builder.Services.AddControllers();
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerDocs();
-builder.Services.AddApplicationConfig();
+builder.Services.AddSwaggerGen(options =>
+{
```
Comment (NdubuisiJr) [inline src/Hng.Web/Program.cs:18]: @Godhanded  This already exists. Check `AddSwaggerDocs` extension methods in `Hng.Web/Extensions/Documentation.cs`. Modify the existing one if need be.

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]
Only two issues to resolve

File: src/Hng.Web/Program.cs (hunk 2 above)
Comment (Godhanded) [reply src/Hng.Web/Program.cs]: Fixed!

**Review by Aniyikaye-Alabi — APPROVED** [review]
Approved

**Review by NdubuisiJr — APPROVED** [review]

**Pull request merged**